src/main/java/com/github/admissionCommittee/dao/GenericDao.java

```diff
@@ -0,0 +1,90 @@
+package com.github.admissionCommittee.dao;
+
+import com.github.admissionCommittee.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+import java.util.List;
+
+public abstract class GenericDao<T> {
+
+    private Class<T> type;
+    private Session session;
+    private Transaction transaction;
+
+    public GenericDao(Class<T> type) {
+        this.type = type;
+    }
+
+    public void create(T instance) {
+        openSessionWithTransaction();
+        session.save(instance);
+        closeSessionWithTransaction();
+    }
+
+    public void update(T instance) {
+        openSessionWithTransaction();
+        session.update(instance);
+        closeSessionWithTransaction();
+    }
+
+    public T get(long id) {
+        openSessionWithTransaction();
+        T instance = session.get(type, id);
+        closeSessionWithTransaction();
+        return instance;
+    }
+
+
+    @SuppressWarnings("unchecked")
+    public List<T> getAll() {
+        openSessionWithTransaction();
+        List<T> instances = (List<T>) session.createQuery("from " + type
+                .getName()).list();
+        closeSessionWithTransaction();
+        return instances;
+    }
+
+    public void delete(long id) {
```

Comment:
double db call and double transaction in one dao method....

Reply:
fixed:
public void delete(long id) {
        openSessionWithTransaction();
        T instance = session.get(type, id);
        if (instance != null) {
            session.delete(instance);
            session.flush();
            closeSession();
        }
    }